consult-org.el

```diff
@@ -0,0 +1,164 @@
+;; Created: 2020
+;; License: GPL-3.0-or-later
+;; Version: 0.6
+;; Package-Requires: ((consult "0.6") (org "9.1") (emacs "26.1"))
+;; Homepage: https://github.com/minad/consult
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Provides a completing-read interface for some Org mode
+;; actions. This is an extra package, to allow lazy loading of Org.
+
+;;; Code:
+
+(require 'consult)
+(require 'org-agenda)
+(require 'org-clock)
+
+(defvar consult-org--history nil)
+
+(defvar consult-org--narrow
+  (append
+   ;; TODO keywords
+   (seq-filter
+    (lambda (it) (<= ?a (car it) ?z))
+    (mapcar (lambda (k)
+                  (list (downcase (string-to-char k)) k
+                        (lambda (cand)
+                          (string-equal k (get-text-property 0 'consult-org-todo cand)))))
+            (mapcan 'cdr org-todo-keywords)))
+   ;; Priorities
+   (mapcar (lambda (c)
+             (list c
+                   (format "Priority %c" c)
+                   (lambda (cand)
+                     (eq c (get-text-property 0 'consult-org-priority cand)))))
+           (number-sequence (max ?A org-highest-priority)
+                            (min ?Z org-lowest-priority)))
+   ;; Outline levels
+   (mapcar (lambda (i)
+             (list (+ i ?0)
+                   (format "Level %i" i)
+                   (lambda (cand)
+                     (>= i (get-text-property 0 'consult-org-level cand)))))
+           (number-sequence 1 9))))
+
+(defun consult-org--narrow ()
+  (cons (lambda (cand)
+          (funcall (nth 2 (assoc consult--narrow consult-org--narrow))
+                   cand))
+        (mapcar (pcase-lambda (`(,c ,s _)) (cons c s))
+                consult-org--narrow)))
+
+(defun consult-org--entries (match scope &rest skip)
+  "Return a list of consult locations from Org entries.
+
+MATCH, SCOPE and SKIP are as in `org-map-entries'."
+  (setq org-outline-path-cache nil)
+  (apply
+   #'org-map-entries
+   (lambda ()
+     (pcase-let ((`(_ ,level ,todo ,prio) (org-heading-components)))
+       (consult--location-candidate
+        (org-format-outline-path (org-get-outline-path t t))
+        (point-marker)
+        (current-line)
+        'consult-org-level level
+        'consult-org-todo todo
+        'consult-org-priority prio)))
+   match scope skip))
```

Comment:
I prefer `consult-org--` for the property prefix. I think I have this consistently over the codebase?

Reply:
At least one property, `consult-location`, is not double dashed. If you decide to change this, may I suggest separating `consult--marker` and `consult--line` to avoid some cars and pcases?

Reply:
No, consult-location is intentionally not double dashed since it is the public metadata for embark-consult.